src/main/scala/org/scalajs/tools/tsimporter/Importer.scala

```diff
@@ -152,13 +152,15 @@ class Importer(val output: java.io.PrintWriter) {
         processDefDecl(classSym, Name.CONSTRUCTOR,
             FunSignature(Nil, params, Some(TypeRefTree(CoreType("void")))), Set.empty[Modifier])
 
-      case PropertyMember(PropertyNameName(name), opt, tpe, mods) if mods(Modifier.Static) =>
+      case PropertyMember(PropertyNameName(name), opt, underlying, mods) if mods(Modifier.Static) =>
         assert(owner.isInstanceOf[ClassSymbol],
             s"Cannot process static member $name in module definition")
         val module = enclosing.getModuleOrCreate(owner.name)
+        val tpe = if (opt) OptionalType(underlying) else underlying
```

Comment:
This line is common between this case and the following. That's a strong indication that it should be factored out in `processPropertyDecl` instead.

Reply:
In addition, once in `processPropertyDecl`, we can apply `if (opt)` directly on the Scala type, rather than the TS type, which means that `OptionalType` becomes unnecessary.

Reply:
Addressed in https://github.com/sjrd/scala-js-ts-importer/pull/105/commits/c6a56b1986f1fca34f9e33ab4afbc0826d808f8b